File: 프로그래머스/2/43165. 타겟 넘버/타겟 넘버.js
function solution(numbers, target) {
    var answer = 0;
    function dfs (index, tot){
        if(index===numbers.length) {
            if(tot===target) ++answer;
            return ;    
        }
        dfs(index+1,tot+numbers[index]);
        dfs(index+1,tot-numbers[index]);
    }
    dfs(0,0)
    return answer;
}

function dfs(idx, accr) {
        if (idx === numbers.length) return accr === target ? ++result : 1
        dfs(idx + 1, accr + numbers[idx])
        dfs(idx + 1, accr - numbers[idx])
    }

File: 프로그래머스/2/43165. 타겟 넘버/타겟 넘버.cpp
#include <string>
#include <vector>

using namespace std;

int answer = 0;

void dfs(int i, int tot, vector<int> numbers, int target){
    if (i==numbers.size()){
        if(tot==target) answer++;
        return ;
        
    }
    dfs(i+1, tot+numbers[i],numbers, target);
    dfs(i+1, tot-numbers[i],numbers, target);
} 

int solution(vector<int> numbers, int target) {
    dfs(0,0, numbers,target); 
    return answer;
}
